public/js/popup.js

```diff
@@ -26,8 +26,33 @@ function locationToString(location) {
     return `[${start}] - [${end}]`;
 }
 
+function getLineFromFile(code, location) {
+    const [[startLine]] = location;
+    const lines = code.split('\n');
+
+    return lines[startLine - 1];
+}
+
+async function fetchCodeLine(event, url, location, cache) {
+    const [target] = event.srcElement.getElementsByClassName('tooltip')
+    target.innerText = "Loading ...";
+
+    const code = cache.get(url) || await fetch(url).then((response) => response.text());
```

Comment:
Using the `url` for cache doesn't work because it will return the same result for multiple warnings on the same **file**. So we have to generate a kind of unique id for each table line and handle the cache with it.

A simple code like this will do the job:
```js
const lineId = Math.random().toString(36).slice(2);
```

Reply:
Yeah. 

My first idea was to cache the file if there are multiple errors on the same file. If I understand properly we should favor the memory size consuming over the network consuming.

I'll update this ^^